### DISCUSSION TITLE
Isolated multiple Termux environments

### DISCUSSION THREAD

**Discussion by lars18th** [termux/termux-packages#11253] [General]
Hi,

First of all, I'm not sure if this is at time already implemented. If you know how to achieve this, them please comment it here. And sorry if this is already commented or documented anywhere.

Second, let me to explain the use case:

- Using the same Android device I want to install _different_ Termux environments. They require to be independent, not necessary in the sense of security. The idea is more similar to _containers_. For example, when you now install Termux in your smartphone, then you can add a lot of different packages and plugins. And that's great. But after that try to _install_ a Linux application that runs inside Termux (and you add for it for example a Termux-GUI interface). And you _really want_ that this application will be _independent_ of your common Termux install. That now is not possible! The reason is how to maintain this Linux application isolated. Without any side effect of what your're touching in your dev Termux environment. For example, when you upgrade your Termux environment, or you're testing something, what happens with this Linux tool? Perhaps it will stop to work at some point.

What's the solution then? Provide a method to **ISOLATE MULTIPLE TERMUX ENVIRONMENTS** inside the same device.
Multiple options could be possible to achieve this:

1. Add support to install Termux multiple times as different applications. This solution will be perfect, except because it will introduce a complexity with the plugins.
2. Add native support to manage independent environements. The problem with this solution is the complexity to manage them.
3. Add a hack tool to Termux to almost provide a layer to encapsulate an alternative environment. Here the problem is the dependency of the inner environement with the outer install. So when you update the main Termux environment the inner will be corrupted.
4. Add support for some type of Termux-containers. But this perhaps goes beyond this project.

Any idea or solution to support this?
I'm open to hearing your comments.

**Comment by 2096779623** [comment]
https://github.com/termux/termux-app/issues/2765

**Reply by lars18th** [reply]
Thank you. I see that https://github.com/termux/termux-app/issues/2765 is related to my request. Even it isn't the same. In my case I can't use the work-profile and I can't use use multiple user accounts. However, I see that others are proposing solutions for _multiple_ Termux installs. And perhaps this generic method could be the most simple solution, because then you can install in multiple Android environments or inside the same. And thinking on the proposed solutions perhaps the patching the $PREFIX suggested could be the most simple: https://github.com/termux/termux-app/issues/2734

Anyway I need to repeat the use case:
- With Termux already installed in my device (for development) I need to "install" one Linux application (based on two binaries and some user scripts, plus a Termux-GUI interface), that I need present it as _independent_ of the regular Termux install. Because I can't mix my Termux environment with this "pseudo-Android-App based on Termux". So in short words: I need a method to **"containerize a Termux App-GUI App"**. 

So, anyone has info about how to compile a full Termux environment with different $PREFIX, including Termux-GUI, plus some packages and put all in one APK ?
